internal/pubsub/pubsub_test.go

```diff
@@ -173,8 +173,15 @@ func TestConcurrentReceivesGetAllTheMessages(t *testing.T) {
 	}
 
 	// Wait for all the goroutines to finish processing all the messages.
-	wg.Wait()
+	for {
+		mu.Lock()
+		if len(receivedMsgs) == howManyToSend {
+			break
```

Comment:
If you hit this line, mu will still be locked.

I like:
```
mu.Lock()
r := len(receivedMsgs)
mu.Unlock()
if r == ...
```

Reply:
Done.